pkg/app/app.go

```diff
@@ -1296,7 +1298,32 @@ func (a *App) apply(r *Run, c ApplyConfigProvider) (bool, bool, []error) {
 		}
 	}
 
-	if releasesToBeDeleted == nil && releasesToBeUpdated == nil {
+	var releasesWithPreApply []state.ReleaseSpec
+	for _, r := range toApplyWithNeeds {
+		release := r
+		if len(release.Hooks) > 0 {
+			for _, hook := range release.Hooks {
+				if slices.Contains(hook.Events, "preapply") {
+					releasesWithPreApply = append(releasesWithPreApply, release)
+					break
+				}
+			}
+		}
+	}
+
+	if len(releasesWithPreApply) > 0 {
+		msg := "Releases with preapply hooks: \n"
+		if infoMsg != nil {
+			msg = fmt.Sprintf("%s%s", *infoMsg, msg)
+		}
+		infoMsg = &msg
+	}
+	for _, release := range releasesWithPreApply {
+		tmp := fmt.Sprintf("%s  %s (%s)", *infoMsg, release.Name, release.Chart)
+		infoMsg = &tmp
+	}
```

Comment:
Can you help explain the logic of infoMsg? @Sajfer

Reply:
There needs to be some type of message in infoMsg for the code on line 1333 to not panic:

```
confMsg := fmt.Sprintf('%s
Do you really want to apply?
  Helmfile will apply all your changes, as shown above.

', *infoMsg)
```


I guess I can either change the message if you want something else, or we have to rewrite the confMsg logic and the debug message that comes after.

Reply:
@Sajfer  why do you comment the codes? Thanks very much.

Reply:
?

Reply:
Was a mistake, It was just done for testing, and accidentally committed.